appserver/web/web-core/src/main/java/org/apache/catalina/connector/CoyoteAdapter.java

```diff
@@ -303,7 +306,16 @@ private void doService(final org.glassfish.grizzly.http.server.Request req,
 //        if (connector.isXpoweredBy()) {
 //            response.addHeader("X-Powered-By", POWERED_BY);
 //        }
-
+        //adding validation to reject request if invalid characters available RFC-9110
+        if (!validateHeaderValues(req)) {
+            String msg = LogFacade.INVALID_HEADER_VALUE_RFC_9110;
```

Comment:
Does this need a variable? Why not use the constant directly?

Reply:
already done!! thanks